### PR TITLE
nginx: net.host.port and net.peer.port should be ints

### PR DESCRIPTION
**Pull request by tobiasstadler**
See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md

### CODE REVIEW

File: instrumentation/nginx/src/script.h [hunk 1]
```diff
@@ -34,19 +34,21 @@ struct NgxCompiledScript {
 
 struct ScriptAttributeDeclaration {
   ScriptAttributeDeclaration(
-    opentelemetry::nostd::string_view attribute, opentelemetry::nostd::string_view script)
-    : attribute(ToNgxString(attribute)), script(ToNgxString(script)) {}
+    opentelemetry::nostd::string_view attribute, opentelemetry::nostd::string_view script, bool convertToInt = false)
```
Comment (seemk) [inline instrumentation/nginx/src/script.h:37]: Nice catch :smile: 

Would an enum make more sense here (and be a bit more readable when reading the declarations list)?

E.g. 
```
  {"http.scheme", "$scheme"},
  {"net.host.port", "$server_port", ScriptAttributeType_Int},
  {"net.peer.ip", "$remote_addr"},
  {"net.peer.port", "$remote_port", ScriptAttributeType_Int},
```

Then the constructor would be something like
```
ScriptAttributeDeclaration(
    opentelemetry::nostd::string_view attribute, opentelemetry::nostd::string_view script, ScriptAttributeType = ScriptAttributeType_String)
```

**Review by seemk — APPROVED** [review]
Thanks! Although I wonder why the traces in tests still have the values as strings :thinking:

**Comment by tobiasstadler** [comment]
> Although I wonder why the traces in tests still have the values as strings 🤔

They don't, sort of. The actually appear as `%{"key" => "net.host.port", "value" => %{"intValue" => "8000"}}`, but the `Map.values(val)` in `attrib` removes the int information.

**Comment by seemk** [comment]
> > Although I wonder why the traces in tests still have the values as strings thinking
> 
> They don't, sort of. The actually appear as `%{"key" => "net.host.port", "value" => %{"intValue" => "8000"}}`, but the `Map.values(val)` in `attrib` removes the int information.

Ah, thanks!

**Comment by tobiasstadler** [comment]
Thank You!

**Pull request merged**